float64 x_rad
float64 y_rad
float64 theta_offset